ext/pgsql/pgsql.c

```diff
@@ -139,88 +284,65 @@ static zend_string *_php_pgsql_trim_message(const char *message)
 		zend_string_release(msgbuf); \
 } \
 
-/* {{{ php_pgsql_set_default_link */
-static void php_pgsql_set_default_link(zend_resource *res)
+static void php_pgsql_set_default_link(zend_object *obj)
 {
-	GC_ADDREF(res);
+	GC_ADDREF(obj);
 
 	if (PGG(default_link) != NULL) {
-		zend_list_delete(PGG(default_link));
-	}
-
-	PGG(default_link) = res;
-}
-/* }}} */
-
-/* {{{ _close_pgsql_link */
-static void _close_pgsql_link(zend_resource *rsrc)
-{
-	PGconn *link = (PGconn *)rsrc->ptr;
-	PGresult *res;
-	zval *hash;
-
-	while ((res = PQgetResult(link))) {
-		PQclear(res);
+		zend_object_release(PGG(default_link));
 	}
-	PQfinish(link);
-	PGG(num_links)--;
 
-	/* Remove connection hash for this link */
-	hash = zend_hash_index_find(&PGG(hashes), (uintptr_t) link);
-	if (hash) {
-		zend_hash_index_del(&PGG(hashes), (uintptr_t) link);
-		zend_hash_del(&EG(regular_list), Z_STR_P(hash));
-	}
+	PGG(default_link) = obj;
 }
-/* }}} */
 
-/* {{{ _close_pgsql_plink */
 static void _close_pgsql_plink(zend_resource *rsrc)
 {
-	PGconn *link = (PGconn *)rsrc->ptr;
-	PGresult *res;
+	if (rsrc->ptr) {
```

Comment:
Why is this if needed now?

Reply:
Unfortunately, I don't remember anymore why I did this. But it's clearly not needed indeed.